### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by amejid**
## Milestone 3: Validations and model specs

In this PR, the following were implemented:-

- Implemented validation for User and Post models
- Added unit specs for all models and methods

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
 class User < ApplicationRecord
-  has_many :posts
-  has_many :likes
-  has_many :comments
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
 
+  validates :name, presence: true
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }
```
Comment (tufoinnkuo10) [inline app/models/user.rb]: - [ ] please add validation for post_counter being an integer

File: app/models/post.rb [hunk 2]
```diff
@@ -5,8 +5,12 @@ class Post < ApplicationRecord
 
   after_save :update_posts_counter
 
-  def return_recent_comments
-    comment.order(created_at: :desc).limit(5)
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :likes_counter, comparison: { greater_than_or_equal_to: 0 }
+  validates :comments_counter, comparison: { greater_than_or_equal_to: 0 }
```
Comment (tufoinnkuo10) [inline app/models/post.rb]: - [ ] kindly add validation for like and comment-counter being an integer

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                      comments_counter: 0)
+      expect(post.valid?).to eq false
+    end
+
+    it 'checks if likes_counter is greater or equal to zero' do
+      post = Post.new(author: @user, title: 'Post communication', text: 'This is my first post', likes_counter: -1,
+                      comments_counter: 0)
+      expect(post.valid?).to eq false
+    end
+
```
Comment (tufoinnkuo10) [inline spec/models/post_spec.rb:39]: - [ ] please add test to check if like and comment counter are integers

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @amejid 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ unit test added

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/models/user.rb (hunk 1 above)
Comment (amejid) [reply app/models/user.rb]: Noted.

File: app/models/post.rb (hunk 2 above)
Comment (amejid) [reply app/models/post.rb]: Noted.

File: spec/models/post_spec.rb (hunk 3 above)
Comment (amejid) [reply spec/models/post_spec.rb:39]: Noted.

**Comment by amejid** [comment]
Thanks for the constructive feedback.

**Review by DeliceLydia — APPROVED** [review]
## Hi @amejid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by amejid** [comment]
Thanks for approving my work.

**Pull request merged**